File: composeApp/src/commonMain/kotlin/com/sdjic/gradnet/presentation/core/model/Post.kt
package com.sdjic.gradnet.presentation.core.model

import com.sdjic.gradnet.presentation.screens.auth.register.model.UserRole

data class Post(
    val postId: String,
    val userId: String,
    val userName: String,
    val userImage: String,
    val content: String,
    val location : String,
    val userRole: UserRole,
    val likesCount: Int,
    val liked: Boolean = false,
    val images: List<String>,
    val createdAt: String
)

File: webApp/src/types/post.ts
export interface PostResponse {
  id: string;
  description: string | null;
  location: string | null;
  images: string[] | null;
  userId: string;
  userName: string | null;
  userRole: string | null;
  userAvatar: string | null;
  likeCount: number;
  createdAt: string;
  updatedAt: string;
  liked: boolean;
}

export interface PostCreateRequest {
  description?: string;
  location?: string;
  images?: string[];
}

export interface LikePostRequest {
  postId: string;
}
